src/main/java/menion/android/whereyougo/gui/extension/activity/CustomActivity.java

```diff
@@ -253,14 +253,34 @@ private void unbindDrawables(View view) {
     }
 
     public static void setLocale(Activity activity, String languageCode) {
-        String lang = languageCode;
+        Locale locale;
+        String lang;
+        Resources resources = activity.getResources();
+        Configuration config = resources.getConfiguration();
+
+        // we need to get language and region first even for default to correctly initialize Locale
         if (languageCode.equals("default")) {
-            lang = Locale.getDefault().getLanguage();
+            lang = Locale.getDefault().toString();
+        } else {
+            lang = new Locale(languageCode).toString();
+        }
+
+        // need to parse language codes with dash and hyphen
+        String[] loc = lang.split("[-_]");
```

Comment:
The language divider is `_`, the `-` is not used.

Reply:
Older android versions may return locale string with dash instead of hyphen, when default system language is requested and that language is regional for upper lang family. I have my emulator for api 21 set to brasilian portuguese and system returns Locale.getDefault like the second line:

```
customOnCreate(), lang: default
setLocale() lang: pt-br
```

Reply:
Interesting, that is not so documented. Then it is fine. :-)